lib/core/api.js

```diff
@@ -494,7 +494,7 @@ api.upload.toPlatform = function(data, sessionInfo, progress, groupId, cb, uploa
         post_dataset_create(uploadItem, function(err, dataset){
           if(_.isEmpty(err)){
             api.log('created dataset');
-            datasetId = _.get(dataset, 'uploadId');
+            datasetId = _.get(dataset, 'data.uploadId');
```

Comment:
I changed the format of the response to the create dataset API. It is a bit more JSONAPI like so that we can return data or errors plus other metadata about the request. I think this makes for a more flexible API.